storage/storage.go

```diff
@@ -0,0 +1,134 @@
+package storage
+
+import (
+	"errors"
+	"github.com/dgraph-io/badger"
+
+	"github.com/iorhachovyevhen/dsss/models"
+)
+
+type Prefix string
+
+const (
+	PrefixSimple Prefix = "simple"
+	PrefixJSON   Prefix = "json"
+	PrefixAudio  Prefix = "audio"
+	PrefixVideo  Prefix = "video"
+)
+
+var DataPrefix = map[models.DataType]Prefix{
+	models.Simple: PrefixSimple,
+	models.JSON:   PrefixJSON,
+	models.Audio:  PrefixAudio,
+	models.Video:  PrefixVideo,
+}
+
+type DataKeeper interface {
+	Add(data models.Data) error
+	Read(key string, data models.Data) error
+	Delete(key string) error
+
+	Close() error
+}
+
+type Storage struct {
+	db      *badger.DB
+	options badger.Options
+}
+
+func NewOptions() badger.Options {
+	return badger.DefaultOptions("")
+}
+
+func NewStorageWithOptions(opts badger.Options) *Storage {
+	return &Storage{
+		db: openDB(opts),
+	}
+}
+
+func NewDefaultStorage(savePath string) *Storage {
+	return &Storage{
+		db: openDB(badger.DefaultOptions(savePath)),
+	}
+}
+
+func openDB(opt badger.Options) *badger.DB {
+	db, err := badger.Open(opt)
+	if err != nil {
+		panic(err)
+	}
+
+	return db
+}
+
+func (s *Storage) Add(data models.Data) error {
```

Comment:
Have you verified that the badger is thread safe for the same key?

Reply:
I'm not sure. but seems that badger resolving that cases

Reply:
I've found the [issue](https://github.com/dgraph-io/badger/issues/756) about it, but it was closed without solve... So, I have the suspicion that it can be not thread-safe for one key.

Reply:
If your test doesn't throw error during inserting with "race" enabled, then all is ok=)

Reply:
I got errors

Reply:
Then you need to add locks=)